### DISCUSSION TITLE
Remote control of iOS device

### DISCUSSION THREAD

**Discussion by joao-borges** [danielpaulus/go-ios#402] [Q&A]
Hi,

I'm evaluating go-ios to replace another solution being used to interact with iOS phones and I can already say that I'm impressed :)

One thing that I am missing is that the ability to remote control the phone. The tool we use provide us with a VNC server to connect. 

I wanted to ask the people here if anyone knows of a possible solution on that direction.

Thank you

**Comment by wellkilo** [comment]
Current go-ios has the building blocks for screen streaming and input, but it is not a turnkey VNC server.

- ios screenshot --stream starts an MJPEG server (default 0.0.0.0:3333; --port changes it).
- Current releases include ios display for a media stream and ios hid / ios ui paths for input automation.
- WebDriverAgent/DeviceKit can provide UI automation, but they require signing/provisioning and are not the same as a generic VNC protocol.

For a simple view-only endpoint:

    ios screenshot --stream --port 3333 --udid DEVICE_UDID

Then view the MJPEG URL from your client, while restricting network access because the stream exposes the device screen.

For interactive remote control, build a small service/UI that combines the stream with supported input calls (or WebDriverAgent). Coordinate mapping, orientation, authentication, reconnects and device locking must be handled by that service. Current CoreDevice HID support is version-limited—its source documents iOS 27+ and a kernel tunnel—so WDA/DeviceKit remains the broadly practical automation route for earlier iOS versions.

The available commands are listed in the current [README help output](https://github.com/danielpaulus/go-ios/blob/c584042907dee35bf80377284ae3aac85990b4d7/README.md#L152-L170), and screenshot streaming is documented in [main.go](https://github.com/danielpaulus/go-ios/blob/c584042907dee35bf80377284ae3aac85990b4d7/main.go#L484-L487). If you require compatibility with an existing VNC viewer, a VNC bridge is still additional software, not a go-ios flag.
